### DISCUSSION TITLE
Setup for official raspberry 7" touchscreen 800x480?

### DISCUSSION THREAD

**Discussion by RTonGithub** [bareboat-necessities/lysmarine_gen#421] [Q&A]
Hi
I am new on bareboat, I have been trying to get me touchscreen working but all my tries end up in that lightdm does not start at boot. 
For example i have tried: 
-Tried changing cvt resolution in config.txt ( hdmi_cvt=800 480 60 but I not sure  
-tried image 0420 and several new installs!
-and a lot more by from various sources on the net, 4 day so far in vain!

/var/log/Xorg.0.log gives:
 Parse error on line 5 of section InputClass in file /usr/share/X11/xorg.conf.d/90-touchinput.conf
        "edt-ft5506" is not a valid keyword in this section.

Any ideas?

**Comment by mgrouch** [comment]
````
Parse error on line 5 of section InputClass in file /usr/share/X11/xorg.conf.d/90-touchinput.conf
"edt-ft5506" is not a valid keyword in this section.
````

it tells you exactly where problem is. File name and line number

**Comment by RTonGithub** [comment]
Yes, 
Of course I was hoping for a more a "what to do" answer, at least a direction where to look for it!
Something with driver it seems, or is it a bookworm, lysmarine issue, or a dts overlay file needed, is the "edt-ft5506" the right driver ?
I have tried to remove the "edt-ft5506" from the file and reboot or run "hot-fixes" but no luck!
I have tried adding dtoverlay=vc4-kms-v3d to config file, no luck
I have tried with the suggested commented lines for other touchscreen in the config files
Actually tried 50 different tests, read 100s of pages on internet to get some idea what to do, but I am fumbling in the dark with a lot of trail and errors!
Maybe if its perfectly clear to everyone what to do, but unfortunately it is not for a NOOB as me!
The raspberry 7" touchscreen has always just worked for me so I hoped for the same here.
Are there someone that have used the raspberry 7" touchscreen and got it working, and have time to give me a hint? 
It would be much appreciated.

**Reply by mgrouch** [reply]
Why don’t you post your file

/usr/share/X11/xorg.conf.d/90-touchinput.conf

here so we could see the line in question?

**Reply by RTonGithub** [reply]
Absolutly, here it is:
user@bareboat:~ $ cat /usr/share/X11/xorg.conf.d/90-touchinput.conf
```
Section "InputClass"
    Identifier "calibration"
    Driver "evdev"
    MatchProduct "10-0038 generic ft5x06 (79)
edt-ft5506
i2c-22-mux (chan_id 1)
bcm2835 (i2c@7e205000)"
    MatchDevicePath "/dev/input/event*"
    Option "EmulateThirdButton" "1"
    Option "EmulateThirdButtonTimeout" "750"
    Option "EmulateThirdButtonMoveThreshold" "30"
EndSection
```
And lightdm --test-mode --debug
```[+0.00s] DEBUG: Logging to /home/user/.cache/lightdm/log/lightdm.log
[+0.00s] DEBUG: Starting Light Display Manager 1.26.0, UID=1000 PID=9893
[+0.00s] DEBUG: Loading configuration dirs from /usr/share/lightdm/lightdm.conf.d
[+0.00s] DEBUG: Loading configuration from /usr/share/lightdm/lightdm.conf.d/01_debian.conf
[+0.00s] DEBUG: Loading configuration from /usr/share/lightdm/lightdm.conf.d/60-lightdm-autologin-greeter.conf
[+0.00s] DEBUG: Loading configuration dirs from /usr/local/share/lightdm/lightdm.conf.d
[+0.00s] DEBUG: Loading configuration dirs from /usr/share/gnome/lightdm/lightdm.conf.d
[+0.00s] DEBUG: Loading configuration dirs from /etc/xdg/lightdm/lightdm.conf.d
[+0.00s] DEBUG: Loading configuration from /etc/lightdm/lightdm.conf.d/lightdm-autologin-greeter.conf
[+0.00s] DEBUG: Loading configuration from /etc/lightdm/lightdm.conf
[+0.00s] DEBUG: Running in user mode
[+0.00s] DEBUG: Registered seat module local
[+0.00s] DEBUG: Registered seat module xremote
[+0.00s] DEBUG: Registered seat module unity
[+0.00s] DEBUG: Using D-Bus name org.freedesktop.DisplayManager
[+0.01s] DEBUG: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
[+0.03s] DEBUG: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ?gio-vfs?
[+0.03s] DEBUG: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
[+0.04s] DEBUG: Monitoring logind for seats
[+0.04s] DEBUG: New seat added from logind: seat0
[+0.04s] DEBUG: Seat seat0: Loading properties from config section Seat:*
[+0.04s] DEBUG: Seat seat0 has property CanMultiSession=no
[+0.04s] DEBUG: Seat seat0: Starting
[+0.04s] DEBUG: Seat seat0: Creating greeter session
[+0.04s] DEBUG: Loading users from org.freedesktop.Accounts
[+0.04s] DEBUG: User /org/freedesktop/Accounts/User1000 added
[+0.05s] DEBUG: Seat seat0: Creating display server of type x
[+0.05s] DEBUG: Seat seat0: Starting local X display
[+0.05s] DEBUG: XServer 0: Logging to /home/user/.cache/lightdm/log/x-0.log
[+0.05s] DEBUG: XServer 0: Writing X server authority to /home/user/.cache/lightdm/run/root/:0
[+0.06s] DEBUG: XServer 0: Launching X Server
[+0.07s] DEBUG: Launching process 9898: /usr/bin/X :0 -seat seat0 -auth /home/user/.cache/lightdm/run/root/:0 -nolisten tcp
[+0.07s] DEBUG: XServer 0: Waiting for ready signal from X server :0
[+0.07s] DEBUG: Acquired bus name org.freedesktop.DisplayManager
[+0.07s] DEBUG: Registering seat with bus path /org/freedesktop/DisplayManager/Seat0
[+0.07s] WARNING: Could not enumerate user data directory /var/lib/lightdm/data: Error opening directory '/var/lib/lightdm/data': Permission denied
[+0.08s] DEBUG: Process 9898 exited with return value 1
[+0.08s] DEBUG: XServer 0: X server stopped
[+0.08s] DEBUG: XServer 0: Removing X server authority /home/user/.cache/lightdm/run/root/:0
[+0.08s] DEBUG: Seat seat0: Display server stopped
[+0.08s] DEBUG: Seat seat0: Stopping session
[+0.08s] DEBUG: Seat seat0: Session stopped
[+0.08s] DEBUG: Seat seat0: Stopping display server, no sessions require it
[+0.08s] DEBUG: Seat seat0: Stopping; greeter display server failed to start
[+0.08s] DEBUG: Seat seat0: Stopping
[+0.08s] DEBUG: Seat seat0: Stopped
[+0.08s] DEBUG: Required seat has stopped
[+0.08s] DEBUG: Stopping display manager
[+0.08s] DEBUG: Display manager stopped
[+0.08s] DEBUG: Stopping daemon
[+0.08s] DEBUG: Exiting with return value 1
```

**Reply by mgrouch** [reply]
MatchProduct line can't have line breaks.

run xinput_calibrator and check what MatchProduct line it generates

**Reply by RTonGithub** [reply]
Tried berfore, but I not sure that I know how!
```
user@bareboat:~ $ sudo xinput_calibrator
[sudo] password for user:
Unable to connect to X server
```

but running ./hot-fixes-install.sh i get the file /usr/share/X11/xorg.conf.d/90-touchinput.conf above

**Reply by mgrouch** [reply]
without sudo

**Reply by RTonGithub** [reply]
Same result, I am affraid!

**Reply by mgrouch** [reply]
Is your X server running? If it didn’t start,
Fix the issue in your /usr/share/X11/xorg.conf.d/90-touchinput.conf

by making MatchProduct value a single line. 
reboot. 

If GUI starts. Start terminal from GUI and run xinput_calibrator from it.

**Reply by RTonGithub** [reply]
After a few trails i suppose you meant like this?
```
Section "InputClass"
    Identifier "calibration"
    Driver "evdev"
    MatchProduct "10-0038 generic ft5x06 (79) edt-ft5506 i2c-22-mux (chan_id 1) bcm2835 (i2c@7e205000)"
    MatchDevicePath "/dev/input/event*"
    Option "EmulateThirdButton" "1"
    Option "EmulateThirdButtonTimeout" "750"
    Option "EmulateThirdButtonMoveThreshold" "30"
EndSection
```
And yes X started! 
and I run the calibration from LXterminal (since I could not see all the rows in the start menu)
Ok, first step done nice, Thank you !
Now I have to find away to make window frames and text smaller!

**Reply by RTonGithub** [reply]
I was a bit a bit hasty, the xinput_calibration does not take my touch input correctly. 
It handles all touches as a touch in the top left corner.
So it is back to the driver issue!

**Reply by mgrouch** [reply]
Try adding Options into that file:

````
Option "Calibration" "0 800 0 480"
````

**Reply by mgrouch** [reply]
Also I’m not convinced that MatchProduct string is correct for your screen. 

find the product:

cat /var/log/Xorg.0.log | grep TOUCH
Then copy and paste product name into MatchProduct section

sudo nano /usr/share/X11/xorg.conf.d/90-touchinput.conf
and save the file.

**Reply by CountOlaf87** [reply]
Okay, so I'm trying this aswell, and I have got it to work, however, I can't see the entire menu. I tried setting hdmi_cvt=800 480 60, but that changes nothing. The resolution looks correct though

**Reply by mgrouch** [reply]
> Okay, so I'm trying this aswell, and I have got it to work, however, I can't see the entire menu. I tried setting hdmi_cvt=[800 480 60](tel:800 480 60), but that changes nothing. The resolution looks correct though

It’s quite low resolution. What you can do is
to run scale-up program. It’s in menu with a blue icon ‘+’. You can ask budgie desktop to autostart it on each boot.

**Comment by RTonGithub** [comment]
Thank you for you suggestions.
```cat /var/log/Xorg.0.log | grep TOUCH
[    20.896] (II) XINPUT: Adding extended input device "10-0038 generic ft5x06 (79)" (type: TOUCHSCREEN, id 10)
```
Seemed to be the same data between the quotes as in the 90-touchinput.conf above.

Option "Calibration" "0 800 0 480" unfortunately did not change the behaviour of xinput_calibrator and the touch input.

I will go back to openplotter, and possibly try later!

**Reply by mgrouch** [reply]
It definitely doesn’t match. Look closely what is in double quotes.

**Comment by mgrouch** [comment]
Closing 

800*480 resolution is a bit too low to be usable
